examples/network_connectivity.py

```diff
@@ -0,0 +1,123 @@
+
+###############################################################################
+# Then we add two proximal drives and simuate the network.
+weights_ampa_p1 = {'L2_basket': 0.08831, 'L2_pyramidal': 0.01525,
+                   'L5_basket': 0.19934, 'L5_pyramidal': 0.00865}
+synaptic_delays_prox = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                        'L5_basket': 1., 'L5_pyramidal': 1.}
+# all NMDA weights are zero; pass None explicitly
+net.add_evoked_drive(
+    'evprox1', mu=26.61, sigma=2.47, numspikes=1, weights_ampa=weights_ampa_p1,
+    weights_nmda=None, location='proximal',
+    synaptic_delays=synaptic_delays_prox, seedcore=4)
+
+# Second proximal evoked drive. NB: only AMPA weights differ from first
+weights_ampa_p2 = {'L2_basket': 0.000003, 'L2_pyramidal': 1.438840,
+                   'L5_basket': 0.008958, 'L5_pyramidal': 0.684013}
+# all NMDA weights are zero; omit weights_nmda (defaults to None)
+net.add_evoked_drive(
+    'evprox2', mu=137.12, sigma=8.33, numspikes=1,
+    weights_ampa=weights_ampa_p2, location='proximal',
+    synaptic_delays=synaptic_delays_prox, seedcore=4)
+
+###############################################################################
+# Data recorded during simulations are stored under
+# :class:`~hnn_core.Cell_Response`. To test multiple network structures, we can
+# create a copy of the original network. The copied network is then simulated.
+net_ERP = net.copy()
+dpl = simulate_dipole(net_ERP, n_trials=1)
+net_ERP.cell_response.plot_spikes_raster()
+
+###############################################################################
+# We can modify the connectivity list to test the effect of different
+# connectivity patterns. For example, we can remove all layer 2 inhibitory
+# connections.
+new_connectivity = [conn for conn in net.connectivity_list
+                    if conn['src_type'] != 'L2Basket']
+net.connectivity_list = new_connectivity
+
+net_remove = net.copy()
+dpl = simulate_dipole(net_remove, n_trials=1)
+net_remove.cell_response.plot_spikes_raster()
+
+###############################################################################
+# That's a lot of spiking! We can additionally add new connections using
+# ``net.add_connection()``. Let's try connecting a single layer 2 basket cell,
+# to every layer 2 pyramidal cell. ``We can utilize net.gid_ranges`` to help
+# find the gids of interest.
+print(net.gid_ranges)
+src_gid = net.gid_ranges['L2_basket'][0]
+target_gids = net.gid_ranges['L2_pyramidal']
+location, receptor = 'distal', 'gabaa'
+
+for target_gid in target_gids:
+    conn = {'src_gid': src_gid,
+            'target_gid': target_gid,
+            'loc': 'soma',
+            'receptor': 'gabab',
+            'delay': 1.0,
+            'weight': 1.0,
+            'threshold': 0.0,
+            'lamtha': 70}
+    net.add_connection(conn)
+
+net_add = net.copy()
+dpl = simulate_dipole(net_add, n_trials=1)
+net_add.cell_response.plot_spikes_raster()
```

Comment:
Looks promising so far! If you can change the example filename so it gets rendered in sphinx gallery that would be great. I'll ponder on the API and the rest of the PR more tomorrow/day after